### PR TITLE
Added event and ref labels

### PR DESCRIPTION
**Pull request by fheinecke**
Here's an example of the changes:
```
gha_workflow_run_time_seconds{event_type="schedule",ref="master",repo="teleport",workflow="dismiss"} 27
gha_step_run_time_seconds{event_type="schedule",job="Dismiss Stale Workflow Runs",ref="master",repo="teleport",step="Dismiss",workflow="dismiss"} 24
gha_step_run_count{conclusion="success",event_type="schedule",job="Dismiss Stale Workflow Runs",ref="master",repo="teleport",step="Dismiss",workflow="dismiss"} 1
gha_job_run_time_seconds{event_type="schedule",job="Dismiss Stale Workflow Runs",ref="master",repo="teleport",workflow="dismiss"} 27
gha_job_run_count{conclusion="success",event_type="schedule",job="Dismiss Stale Workflow Runs",ref="master",repo="teleport",workflow="dismiss"} 1
```

Another, merge group and branch/v14:
```
gha_step_run_count{conclusion="success",event_type="merge_group",job="Integration Tests (Non-root)",ref="branch/v14",repo="teleport",step="Checkout Teleport",workflow="integration-tests-non-root"} 1
```

### CODE REVIEW

File: collector.go [hunk 1]
```diff
@@ -220,34 +221,66 @@ func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
...
+		}
+
+		for _, pr := range run.PullRequests {
+			prBaseBranch := pr.GetBase()
+			if prBaseBranch == nil {
+				continue
+			}
+
+			if prBaseBranch.GetSHA() == headSha {
+				return headSha
```
Comment (camscale) [inline collector.go]: I dont think we want to return a SHA for the ref - it really needs to be a branch name or nothing. At the very least this should come after the headBranch check so we return the branch name if it matches before we care about the SHA, but I still think the SHA is wrong for a timeseries label as SHAs are typically unique and you don't want unique values in timeseries labels (its not much of a "series" then).

File: collector.go [hunk 2]
```diff
@@ -220,34 +221,66 @@ func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
...
+		headSha := run.GetHeadSHA()
+		headBranch := run.GetHeadBranch()
+
+		if headSha == "" && headBranch == "" {
+			return "error-missing-head-ref"
+		}
+
+		for _, pr := range run.PullRequests {
+			prBaseBranch := pr.GetBase()
+			if prBaseBranch == nil {
```
Comment (camscale) [inline collector.go]: this condition is not needed - the purpose of the `Get*()` accessor functions is to handle `nil` for you.

File: collector.go [hunk 3]
```diff
@@ -220,34 +221,66 @@ func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
...
+			if prBaseBranch.GetSHA() == headSha {
+				return headSha
+			}
+
+			if prBaseBranch.GetRef() == headBranch {
+				return headBranch
+			}
+		}
+
+		return "error-no-matching-pr"
```
Comment (camscale) [inline collector.go]: I dont think this is an error - the value should either be empty or `"unknown"`. Not every workflow run with a pull request event will be able to be tied back to a PR - the description of the `pull_requests` field in the ["Get a workflow run"](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#get-a-workflow-run) docs says

> Pull requests that are open with a `head_sha` or `head_branch` that matches the workflow run. The returned pull requests do not necessarily indicate pull requests that triggered the run.

So if the pull request has closed at the time we query the the workflow runs, we may not get a PR. That's not an error, just bad timing, so I don't think the value should indicate an error.

File: collector.go [hunk 4]
```diff
@@ -220,34 +221,66 @@ func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
...
+	}
+
+	headBranch := run.GetHeadBranch()
```
Comment (camscale) [inline collector.go:284]: I'm wondering if this is what we want for merge_group events - I assume for merge_group that the pull requests would be valid in the run, so perhaps the above logic for pull requests should also apply to merge group events. 

I'm still waiting to find a merge_group PR with `pull_requests` non-empty - but an alternative is to parse the branch out of the head branch which is of the form "gh-readonly-queue/master/pr-39050-deeadc4b4cee7de73be618a1b612955ea6006d0b"

File: collector.go (hunk 3 above)
Comment (fheinecke) [reply collector.go]: Do you think "error-no-matching-open-pr" would be better, or maybe the head branch? I'm attempting to prefix all the "this is not what you would expect and should be filtered out in queries" values with the same prefix.

> Pull requests that are **open**

I really hate this API...

File: collector.go (hunk 4 above)
Comment (fheinecke) [reply collector.go:284]: Added logic for this - PTAL.

**Review by camscale — APPROVED** [review]
LGTM. Let's give this a try and see what the metrics look like

**Pull request merged**